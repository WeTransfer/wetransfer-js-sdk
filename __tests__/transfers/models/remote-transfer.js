const { RemoteTransfer } = require('../../../src/transfers/models');

describe('RemoteTransfer model', () => {
  let transfer = null;
  beforeEach(() => {
    transfer = new RemoteTransfer({
      id: 'random-hash',
      message: 'Little kittens',
      state: 'uploading',
      url: 'https://we.tl/t-random-hash',
      files: [
        {
          id: 'random-hash',
          name: 'kittie.gif',
          type: 'file',
          size: 1024,
        },
      ],
      expires_at: '2018-01-01T00:00:00Z',
    });
  });

  describe('contructor', () => {
    it('should create a normalized transfer', () => {
      expect(transfer).toMatchSnapshot();
    });
  });
});
