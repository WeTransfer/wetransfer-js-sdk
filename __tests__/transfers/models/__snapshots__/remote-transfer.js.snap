// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoteTransfer model contructor should create a normalized transfer 1`] = `
RemoteTransfer {
  "expires_at": 2018-01-01T00:00:00.000Z,
  "files": Array [
    Object {
      "id": "random-hash",
      "multipart": undefined,
      "name": "kittie.gif",
      "size": 1024,
      "type": "file",
    },
  ],
  "id": "random-hash",
  "message": "Little kittens",
  "state": "uploading",
  "url": "https://we.tl/t-random-hash",
}
`;
